docs/src/references/c8y-configuration-management.md

```diff
@@ -162,4 +162,235 @@ Note that:
+
+Example:
+
+```bash
+    $ tree /etc/tedge/c8y/
+    /etc/tedge/c8y/
+    ├── c8y-configuration-plugin.toml
+    ├── child1
+    │   └── c8y-configuration-plugin.toml
+    └── child2
+        └── c8y-configuration-plugin.toml    
+```
+
+The `c8y_configuration_plugin` adds per child-device the file `/etc/tedge/c8y/<childid>/c8y-configuration-plugin.toml` implicitely to the child-device's config listed. So the list can always be configured from the cloud. The `type` for this self configuration file is `c8y-configuration-plugin`.
+
+Each time a child-device's TOML is modified (e.g. due to an incoming MQTT message, an incoming new configuration snapshot from the cloud, or a modification by a local process) the `c8y_configuration_plugin` takes care to define all necessary capabilities to the coresponding cloud's child-device twin. These are:
+  - declaring _supported operations_ for configuration management: `c8y_UploadConfigFile` and `c8y_DownloadConfigFile`
+  - declaring provided _configuration types_
+
+**Declaring 'supported operations'**
+
+To declare supported operations the `c8y_configuration_plugin` uses thin-edge's _Supported Operations API_. Therefore the `c8y_configuration_plugin` creates for each child-device two files under `/etc/tedge/operations/c8y/<childid>`.
+
+Example, for child-device with childid `child1`:
+
+```
+/etc/tedge/operations/c8y/child1/c8y_UploadConfigFile
+/etc/tedge/operations/c8y/child1/c8y_DownloadConfigFile
+```
+
+As soon as those files are created, thin-edge's _Supported Operations API_ takes care to send according supported operation declarations to cloud's child-device twins (see [documentation Supported Operations](../tutorials/supported_operations.md#supported-operations-for-child-devices)).
+
+**Declaring 'provided configuration types'**
+
+For all configuration `types` provided by the external device, the plugin sends an MQTT message to C8Y. Thereby all `types` will be combined in one single message as below:
+  - topic: `c8y/s/us/<childid>`
+  - payload: `119,<type 1>,<type 2>,<type 3>,...`<br/>
+    Example: `119,foo.conf,bar.conf`
+
+Note that the `c8y_configuration_plugin` does **not** create any child-device twin in the cloud. Instead the clouds child-device twins must be created upfront.
+
+
+## Fetching/Pushing configuration file from/to the external device
+
+To fetch/push configuration files to/from external devices, the `c8y_configuration_plugin` makes use of file-transfer HTTP APIs of the `tedge_agent`.
+  
+The HTTP filetransfer feature of the `tedge_agent` provides the service to transfer files from external devices to the local filesystem of the thin-edge device's, and vice versa.
+
+  * `tedge_agent` serves PUT, GET and DELETE requests for temporary file upload, download and removal
```

Comment:
The external-device needs anyway to know the IP address of the thin-edge device, since it must connect the MQTT broker. Obvisouly the external-device needs to provide a way to the device operator to configure that (optionally also credentials/certificate, if thin-edge's broker is somehow secured).

Even that is a part the developer/owner of the external-device must consider thats a good information to add to the user journeys / child-device model of our overall child-device reference doc. I will take care to add that.